examples/net/listener.pony

```diff
@@ -8,17 +9,26 @@ class Listener is TCPListenNotify
   var _host: String = ""
   var _service: String = ""
   var _count: USize = 0
+  let _root: AmbientAuth
 
-  new create(env: Env, ssl: Bool, limit: USize) =>
+  new create(env: Env, ssl: Bool, limit: USize) ? =>
     _env = env
+    _root = match env.root
+	  | let r: AmbientAuth => r
+	  else
+	    error
+	  end
```

Comment:
But more fundamentally, why not just require the `AmbientAuth` here, instead of the entire `Env`? Then you can avoid the need to carry the `Env` to this point, *and* avoid the possibility of an `error` here.